rust/src/presentation/mod.rs

```diff
@@ -0,0 +1,188 @@
+use std::{collections::HashMap, str::FromStr, sync::Arc};
+
+use serde::de::{Deserialize, IntoDeserializer};
+use ssi::{
+    claims::{data_integrity::AnyProtocol, MessageSignatureError, SignatureEnvironment},
+    crypto::AlgorithmInstance,
+    dids::{AnyDidMethod, VerificationMethodDIDResolver},
+    json_ld::{iref::UriBuf, ContextLoader, IriBuf},
+    prelude::{AnySuite, CryptographicSuite, ProofOptions},
+    verification_methods::{protocol::WithProtocol, MessageSigner, ProofPurpose},
+};
+
+pub use error::*;
+
+use crate::{
+    credential::{ParsedCredential, ParsedCredentialInner},
+    crypto::CryptoCurveUtils,
+    oid4vp::PresentationSigner,
+};
+
+mod error;
+
+#[derive(Debug, Clone, uniffi::Object)]
+pub struct JsonLdPresentationBuilder {
+    pub(crate) id: String,
+    pub(crate) holder: String,
+
+    pub(crate) proof_purpose: ProofPurpose,
+    pub(crate) challenge: Option<String>,
+    pub(crate) domain: Option<String>,
+
+    pub(crate) signer: Arc<Box<dyn PresentationSigner>>,
+    pub(crate) context_map: Option<HashMap<String, String>>,
+}
+
+#[uniffi::export]
+impl JsonLdPresentationBuilder {
+    #[uniffi::constructor(name = "new")]
+    fn new(
+        id: String,
+        holder: String,
+
+        proof_purpose: String,
+        challenge: Option<String>,
+        domain: Option<String>,
+
+        signer: Box<dyn PresentationSigner>,
+        context_map: Option<HashMap<String, String>>,
+    ) -> Arc<Self> {
+        let proof_purpose: Result<ProofPurpose, serde::de::value::Error> =
+            ProofPurpose::deserialize(proof_purpose.into_deserializer());
+        Self {
+            id,
+            holder,
+            proof_purpose: proof_purpose.unwrap(),
+            challenge,
+            domain,
+            signer: Arc::new(signer),
+            context_map,
+        }
+        .into()
+    }
+
+    pub async fn issue_presentation(
```

Comment:
Might make more sense to just implement this only for the credential types  that can be ld presentations